### PR TITLE
Refactor the remaining pieces of the sign-in process

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

1. The check for invited user email vs expected email was already moved to register.py in a previous PR, so it is in place and working.
2. Added many debug statements so in the future we can just temporarily set the log level on production to debug.
3. Added tests for accepting invite and for failing due to email address mismatch
4. Added tests for encoding/decoding invite data
5. removed old code and tests pointing to things (/register) that don't exist anymore.

## Security Considerations

1. the debug statements dump a lot of stuff including user's email addresses.  The thinking is we typically need this info to debug and we would only enable it on production during an urgent debug session, for a short period of time.  I think there is already a ticket to make some way to automatically scrub logs and partially mask email addresses etc, but that doesn't exist yet.
2. We are passing the invite data in the URL and it is not particularly hard to decode it.  Amongst this info is the invitee's email address.  Do we have to actually encrypt this information?  I feel like we are only sending it to the invitee's email inbox, and from there they are clicking on the link and going to login.gov, etc.  It's not clear to me that it's such a big risk to leave it as is with just base64 encoding.

### CODE REVIEW

**Review by ccostino — COMMENTED** [review]
Awesome, thanks @terrazoon!  On the whole this looks good to me and it appears like it'll at least address the issue, though I still need to test locally.  I had a few comments and questions that I've left in the mean time as I continue going through things and others get a chance to take a look.

By all accounts it looks like this takes care of everything in https://github.com/GSA/notifications-admin/issues/1459 too, except possibly some E2E test things with the Login.gov integration, but that's fine to save for another issue (which we should have, I'll check the board).

Nicely done with the test cleanup and updates as well, thank you! 🎉

File: app/main/views/sign_in.py [hunk 1]
```diff
@@ -146,68 +144,13 @@ def sign_in():
...
-
-            if user.state == "pending":
-                return redirect(
-                    url_for("main.resend_email_verification", next=redirect_url)
-                )
-
-            if user.is_active:
-                if session.get("invited_user_id"):
-                    invited_user = InvitedUser.from_session()
-                    if user.email_address.lower() != invited_user.email_address.lower():
```
[the remark this answers is not in the document]
Comment (terrazoon) [reply app/main/views/sign_in.py:184]: It's been moved over to register.py and is currently working, I've tested it.

File: app/models/user.py [hunk 2]
```diff
@@ -141,7 +141,9 @@ def set_permissions(self, service_id, permissions, folder_permissions, set_by_id
         )
 
     def logged_in_elsewhere(self):
-        return session.get("current_session_id") != self.current_session_id
+        # This check is deprecated due to the transition to using login.gov.
+        return False
+        # return session.get("current_session_id") != self.current_session_id
```
[the remark this answers is not in the document]
Comment (terrazoon) [reply app/models/user.py:146]: It's not used anywhere else.

File: app/main/views/sign_in.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ccostino) [reply app/main/views/sign_in.py:184]: I thought it existed elsewhere, I just couldn't remember where.  Thanks!

File: app/models/user.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccostino) [reply app/models/user.py:146]: If this isn't used anywhere else, let's get rid of it entirely. 🪓

**Review by ccostino — COMMENTED** [review]
One final change and I think we're good to merge this to staging for final testing - all looked good to me locally, but I haven't been able to replicate the issue locally it seems.  Only in staging or production.

**Review by ccostino — APPROVED** [review]
Thanks, @terrazoon!  Let's get this into staging for some more testing!

**Pull request merged**